third_party/CMakeLists.txt

```diff
@@ -93,3 +93,45 @@ if(VELOX_ENABLE_ARROW)
                                            ${ARROW_LIBDIR}/libparquet.a)
 
 endif()
+
+if(VELOX_ENABLE_QPL)
+  message(STATUS "Building QPL from source")
+  set(QPL_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/qpl_ep/install")
+  set(QPL_STATIC_LIB_NAME
+      ${CMAKE_STATIC_LIBRARY_PREFIX}qpl${CMAKE_STATIC_LIBRARY_SUFFIX})
+  set(QPL_STATIC_LIB "${QPL_PREFIX}/lib/${QPL_STATIC_LIB_NAME}")
+
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ldl -laccel-config -L/usr/lib64")
+
+  set(QPL_CMAKE_ARGS
+      -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+      -DCMAKE_INSTALL_LIBDIR=${QPL_PREFIX}/lib
+      -DCMAKE_INSTALL_PREFIX=${QPL_PREFIX}
+      -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+      -DEFFICIENT_WAIT=OFF
+      -DQPL_BUILD_TESTS=OFF
+      -DCMAKE_C_COMPILER=gcc
+      -DCMAKE_CXX_COMPILER=g++
+      -DQPL_BUILD_EXAMPLES=OFF
+      -DQPL_LIB=ON)
+
+  ExternalProject_Add(
```

Comment:
I see that qpl uses CMake, in that case please use [FetchContent ](https://cmake.org/cmake/help/latest/module/FetchContent.html). That way you won't have to add the targets manually.